app/services/parser.rb

```diff
@@ -0,0 +1,64 @@
+require 'open-uri'
+
+class Parser
+  HEADLESS_RUN = true
+  DELAY_VAL = 3
+
+  def start(count=2)
+    if HEADLESS_RUN
+      headless = Headless.new
+      headless.start
+    end
+    
+    doc = Nokogiri::HTML(open('https://www.avito.ru/irkutsk/kvartiry/sdam/na_dlitelnyy_srok'));
+    
+    doc.css('h3.title a').each_with_index do |link, index|
+      break if ++index == count
+      delay if index != 0
+
+      browser = Watir::Browser.new :firefox
+      item = {}
+      item_doc = Nokogiri::HTML(open("https://www.avito.ru#{link['href']}"));
+      item[:renter] = item_doc.css('#seller').text.gsub(/\w|\s/,'')
+      item[:price] = item_doc.css('.p_i_price').text.gsub(/\D/,'')
+      item[:description] = item_doc.css('div#desc_text').text
+      item[:city] = item_doc.css('div#map').text
+      item[:address] = item_doc.css('span#toggle_map').text
+      item[:id_ad] = item_doc.css('span#toggle_map').text
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.